### ISSUE TITLE
Logstash is mangling multiline logs

### ISSUE THREAD

**Issue by moebiuseye** [elastic/logstash#4161]
I've been trying to figure out this very bizarre difference between my input and output: 

First, here's the configuration I'm using (couldn't be simpler): 

```ruby
input {
  syslog {
    port => 5140
  }
}
output {
  elasticsearch { hosts => ["localhost:9200"] }
}
```

And heres is a diff between my input and my output:

```diff
2,8c2,7
< com.zimbra.common.service.ServiceException: no valid authtoken present
< ExceptionId:qtp1778535015-85870:http://[REDACTED]/zimbra/h/search;jsessionid=jj6olwou79qg1sad9kmmwy55a?initial=true&mesg=welcome:1447146302691:b4c53833bc794a50
< Code:service.AUTH_REQUIRED
<         at com.zimbra.common.service.ServiceException.AUTH_REQUIRED(ServiceException.java:322)
<         at com.zimbra.cs.taglib.ZJspSession.getZMailbox(ZJspSession.java:463)
<         at com.zimbra.cs.taglib.tag.ZimbraSimpleTag.getMailbox(ZimbraSimpleTag.java:29)
< 
---
> com.zimbra.common.service.Servic no valid authtoken present
> qtp1778535015-85870:http://[REDACTED]/zimbra/h/search;jsessionid=jj6olwou79qg1sad9kmmwy55a?initial=true&mesg=welcome:1447146302691:b4c53833bc794a50
> service.AUTH_REQUIRED
>     at com.zimbra.common.service.ServiceException.AUTH_REQUIRED(ServiceException.java:322)
>     at com.zimbra.cs.taglib.ZJspSession.getZMailbox(ZJspSession.java:463)
>     at com.zimbra.cs.taglib.tag.ZimbraSimpleTag.getMailbox(ZimbraSimpleTag.java:29)

```

Before anyone asks and if it's any relevant, I'm running:

* Centos 7
* Logstash 2.0.0
* Ruby ruby 2.0.0p598 (2014-11-13)
* Java(TM) SE Runtime Environment (build 1.8.0_66-b17)

**Comment by moebiuseye** [comment]
Here's a sample generated using the file output:

```json
{"message":"WARN  [qtp1778535015-86257:http://192.168.2.103:8090/zimbra/h/search;jsessionid=uh37ml0rpefh1do1layfvbmpq?initial=true&mesg=welcome] [] webclient - no valid authtoken present","@version":"1","@timestamp":"2015-11-10T09:51:06.000Z","host":"192.168.2.103","priority":132,"timestamp":"Nov 10 10:51:06","logsource":"localhost.localdomain","program":"mailboxd","severity":4,"facility":16,"facility_label":"local0","severity_label":"Warning"}
{"message":"com.zimbra.common.service.Servic no valid authtoken present","@version":"1","@timestamp":"2015-11-10T09:51:06.000Z","host":"192.168.2.103","priority":132,"timestamp":"Nov 10 10:51:06","logsource":"localhost.localdomain","severity":4,"facility":16,"facility_label":"local0","severity_label":"Warning"}
{"message":"qtp1778535015-86257:http://192.168.2.103:8090/zimbra/h/search;jsessionid=uh37ml0rpefh1do1layfvbmpq?initial=true&mesg=welcome:1447149066940:b4c53833bc794a50","@version":"1","@timestamp":"2015-11-10T09:51:06.000Z","host":"192.168.2.103","priority":132,"timestamp":"Nov 10 10:51:06","logsource":"localhost.localdomain","program":"ExceptionId","severity":4,"facility":16,"facility_label":"local0","severity_label":"Warning"}
{"message":"service.AUTH_REQUIRED","@version":"1","@timestamp":"2015-11-10T09:51:06.000Z","host":"192.168.2.103","priority":132,"timestamp":"Nov 10 10:51:06","logsource":"localhost.localdomain","program":"Code","severity":4,"facility":16,"facility_label":"local0","severity_label":"Warning"}
{"message":"    at com.zimbra.common.service.ServiceException.AUTH_REQUIRED(ServiceException.java:322)","@version":"1","@timestamp":"2015-11-10T09:51:06.000Z","host":"192.168.2.103","priority":132,"timestamp":"Nov 10 10:51:06","logsource":"localhost.localdomain","severity":4,"facility":16,"facility_label":"local0","severity_label":"Warning"}
{"message":"    at com.zimbra.cs.taglib.ZJspSession.getZMailbox(ZJspSession.java:463)","@version":"1","@timestamp":"2015-11-10T09:51:06.000Z","host":"192.168.2.103","priority":132,"timestamp":"Nov 10 10:51:06","logsource":"localhost.localdomain","severity":4,"facility":16,"facility_label":"local0","severity_label":"Warning"}
{"message":"    at com.zimbra.cs.taglib.tag.ZimbraSimpleTag.getMailbox(ZimbraSimpleTag.java:29)","@version":"1","@timestamp":"2015-11-10T09:51:06.000Z","host":"192.168.2.103","priority":132,"timestamp":"Nov 10 10:51:06","logsource":"localhost.localdomain","severity":4,"facility":16,"facility_label":"local0","severity_label":"Warning"}
{"message":"    at com.zimbra.cs.taglib.tag.GetMailboxTag.doTag(GetMailboxTag.java:57)","@version":"1","@timestamp":"2015-11-10T09:51:06.000Z","host":"192.168.2.103","priority":132,"timestamp":"Nov 10 10:51:06","logsource":"localhost.localdomain","severity":4,"facility":16,"facility_label":"local0","severity_label":"Warning"}
```

**Comment by moebiuseye** [comment]
I see something odd on the second and third lines, here: 

```json
... "program":"ExceptionId" ...
... "program":"Code" ...
```

**Comment by moebiuseye** [comment]
Also, as asked by Rumbles on IRC, when forwarding to an rsyslog server, the logs are fine. The problem comes from logstash, not syslog. 

Here's a sample from the logs forwarded through rsyslog:

```logs
Nov 10 14:09:23 localhost.localdomain mailboxd: WARN  [qtp1778535015-87887:http://[REDACTED]/zimbra/h/search;jsessionid=1n54k9080c1131tf7jha1m68tt?initial=true&mesg=welcome] [] webclient - no valid authtoken present
Nov 10 14:09:23 localhost.localdomain com.zimbra.common.service.Servic no valid authtoken present
Nov 10 14:09:23 localhost.localdomain ExceptionId: qtp1778535015-87887:http://[REDACTED]/zimbra/h/search;jsessionid=1n54k9080c1131tf7jha1m68tt?initial=true&mesg=welcome:1447160963471:b4c53833bc794a50
Nov 10 14:09:23 localhost.localdomain Code: service.AUTH_REQUIRED
Nov 10 14:09:23 localhost.localdomain     at com.zimbra.common.service.ServiceException.AUTH_REQUIRED(ServiceException.java:322)
Nov 10 14:09:23 localhost.localdomain     at com.zimbra.cs.taglib.ZJspSession.getZMailbox(ZJspSession.java:463)
Nov 10 14:09:23 localhost.localdomain     at com.zimbra.cs.taglib.tag.ZimbraSimpleTag.getMailbox(ZimbraSimpleTag.java:29)
Nov 10 14:09:23 localhost.localdomain     at com.zimbra.cs.taglib.tag.GetMailboxTag.doTag(GetMailboxTag.java:57)
```

**Comment by suyograo** [comment]
Fixed in v2.1.0 of LS

**Comment by moebiuseye** [comment]
No it isn't fixed in v2.1.0 of LS

**Comment by moebiuseye** [comment]
I've figured out why this wasn't working properly and it turns out it's an issue with date formatting. Here's how I solved this: 

~~~
$template RFC3339fmt,"<%PRI%>%TIMESTAMP:::date-rfc3339% %HOSTNAME% %syslogtag%%msg%"
*.* @@10.0.0.1:514;RFC3339fmt
~~~
